### ISSUE TITLE
The `equippable` component does not work in nbt

### ISSUE THREAD

**Issue by Slarof** [PluginBugs/Issues-ItemsAdder#4192]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
- [X] I've looked for already existing issues on the [Issue Tracker](https://github.com/PluginBugs/Issues-ItemsAdder/issues) and haven't found any.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.
- [X] I tested this with just ItemsAdder and its dependencies and with a vanilla client of the same version as the Server.

### Discord Username (optional)

vova_erofeev

### What happened?

I wanted to add armor based on the `equippable` component via nbt, but it doesn't work

- When putting on an item, its armor texture is not displayed, although it is specified in the value `model`. Instead, the texture of the armor model that is peculiar to the item is displayed (for example, chainmail boots)
- When the value of `slot` is set to the specified slot, an item (for example, a stick) cannot be put on this slot

### Steps to reproduce the issue

1. Opening the configuration of the item
2. Add the nbt string with the `equipable` component and save
3. Start the server, give an item with the `equipable` component and notice that it does not work

### Server version

Current Purpur Version: 1.21.3-2349-2b6f273 (MC: 1.21.3)*
* You are running the latest version

### ItemsAdder Version

ItemsAdder version 4.0.5-beta-5
Customized Minecraft without client mods!
Website: devs.beer
Author: LoneDev

### ProtocolLib Version

ProtocolLib version 5.4.0-SNAPSHOT-736
Provides read/write access to the Minecraft protocol.
Authors: dmulloy2 and comphenix

### LoneLibs Version

LoneLibs version 1.0.61
Collection of libs used by my plugins
Authors: LoneDev and others

### Full Server Log

https://mclo.gs/3yUDScv

### Problematic items yml configuration file (optional)

```yaml
info:
  namespace: test

items:
  fake_diamond_boots:
    enabled: true
    display_name: "Фейковые алмазные ботинки"
    nbt: "{components: {
    \"minecraft:equippable\": {
      slot: \"feet\",
      model: \"minecraft:diamond\",
      allowed_entities: \"minecraft:player\"}}}"
    resource:
      material: CHAINMAIL_BOOTS
      generate: true
      textures:
      - item/fake_diamond_boots.png
```

### Screenshots/Videos (you can drag and drop files or paste links)

YouTube: https://youtu.be/KJB78tHcuJE
- In the video, chainmail boots with the texture of diamond boots do not want to change their armor model to another one

**Comment by Slarof** [comment]
Yes, I know about it: https://github.com/PluginBugs/Issues-ItemsAdder/issues/3980

- This is most likely different, and it should work in nbt anyway.

**Comment by LoneDev6** [comment]
Should be fixed in 4.0.5
